DOC: Migration guide should mention complex number API changes

### Describe the issue:

I'm helping the h5py project migrate to Cython 3 and while doing so also tested with numpy 2.0 (`main`). I discovered some compilation errors which you can read in detail here:

https://github.com/h5py/h5py/issues/2348

Basically h5py uses a macro from the HDF5 C++ library to get the offset into a struct for the "real" and "imag" members, but now complex types in numpy's C API are not structs (https://github.com/numpy/numpy/pull/24085). So I'm wondering what the best long term solution is to work around this. The h5py code doing this looks like this:

```
 #define h5py_offset_n64_real (HOFFSET(npy_complex64, real)) 
```

### Reproduce the code example:

```python
Not a python problem so no code example. See above for compilation information. Sorry for filing this as a bug but the "support" page on the docs lead to gitter, reddit, and IRC which all seemed inactive. This seems like a possibly specific case to h5py so I didn't really want to put it on stackoverflow either. And I wasn't really sure what else to categorize this as.
```


### Error message:

```shell
/tmp/pip-req-build-3skh7xs0/h5py/h5t.c: In function ‘__pyx_f_4h5py_3h5t__c_complex’:
      /tmp/pip-req-build-3skh7xs0/h5py/api_compat.h:38:31: error: request for member ‘real’ in something not a structure or union
         38 | #define h5py_offset_n64_real (HOFFSET(npy_complex64, real))
            |                               ^~~~~~~
      /tmp/pip-req-build-3skh7xs0/h5py/h5t.c:26702:21: note: in expansion of macro ‘h5py_offset_n64_real’
      26702 |     __pyx_v_off_r = h5py_offset_n64_real;
            |                     ^~~~~~~~~~~~~~~~~~~~
```


### Python and NumPy Versions:

Python 3.11
Numpy 2.0 (main)

### Runtime Environment:

_No response_

### Context for the issue:

Currently affecting h5py compatibility with numpy 2.0.

--- a/doc/source/numpy_2_0_migration_guide.rst
+++ b/doc/source/numpy_2_0_migration_guide.rst
@@ -125,6 +125,25 @@ case you need to pass ``NPY_RAVEL_AXIS`` instead of ``NPY_MAXDIMS``.
 ``NPY_RAVEL_AXIS`` is defined in the ``npy_2_compat.h`` header and runtime
 dependent (mapping to 32 on NumPy 1.x and ``-2147483648`` on NumPy 2.x).
 
+Complex types - Underlying type changes
+---------------------------------------
+
+The underlying C types for all of the complex types have been changed to use
+native C99 types. While the memory layout of those types remains identical
+to the types used in NumPy 1.x, the API is slightly different, since direct
+field access (like ``c.real`` or ``c.imag``) is no longer possible.
+
+It is recommended to use the functions `npy_creal` and `npy_cimag` (and the
+corresponding float and long double variants) to retrieve
+the real or imaginary part of a complex number, as these will work with both
+NumPy 1.x and with NumPy 2.x. New functions `npy_csetreal` and `npy_csetimag`,
+along with compatibility macros `NPY_CSETREAL` and `NPY_CSETIMAG` (and the
+corresponding float and long double variants), have been
+added for setting the real or imaginary part.
+
+The underlying type remains a struct under C++ (all of the above still remains
+valid).
+
 
 Namespace changes
 =================

